rtgui/previewhandler.cc

```diff
@@ -204,27 +204,19 @@ Glib::RefPtr<Gdk::Pixbuf> PreviewHandler::getRoughImage (int x, int y, int w, in
             h = image->getHeight() * totalZoom;
         }
 
-        int ix = x * zoom;
-        int iy = y * zoom;
+        x *= zoom;
+        y *= zoom;
 
-        if (ix < 0) {
-            ix = 0;
+        if ((x + w) / totalZoom > previewImg->get_width()) {
```

Comment:
I would try to avoid depending on algebraic identities with floating-point arithmetic, i.e. I  would prefer something like
````
w = std::min(w, previewImage->get_width() * totalZoom - x);
````
or even
````
w = std::min(w, std::max(previewImage->get_width() * totalZoom - x, 0));
````
if we are being pedantic.